packages/otp2-tile-overlay/src/index.tsx

```diff
@@ -150,7 +150,7 @@ const OTP2TileLayerWithPopup = ({
             configCompanies={configCompanies}
             entity={{ ...clickedEntity, id: clickedEntity?.id || clickedEntity?.gtfsId }}
             setLocation={setLocation ? (location) => { setClickedEntity(null); setLocation(location) } : null}
-            setViewedStop={setViewedStop}
+            setViewedStop={setViewedStop ? (stop) => { setClickedEntity(null);setViewedStop(stop) } : null}
```

Comment:
hate to do this to ya, but can you put this in its own callback function with a `useCallback`?

Reply:
When is this reused though? The stop is always changing, so the function would constantly be getting regenerated anyway!

Reply:
You're right, good point